src/main/java/com/salesforce/cdp/queryservice/util/Utils.java

```diff
@@ -37,4 +38,86 @@ private Utils() {
+        int bytesNeedingReEncode = 0;
+
+        // As per the URLEncode.encode() documentation we implement the byte array version here
+        // (to keep us from forming immutable strings)
+        // The alphanumeric characters "a" through "z", "A" through "Z" and "0" through "9" remain the same.
+        // The special characters ".", "-", "*", and "_" remain the same.
+        // The space character " " is converted into a plus sign "+".
+        // All other characters are unsafe and are first converted into one or more bytes using some encoding scheme. Then each byte is represented by the 3-character string "%xy", where xy is the two-digit hexadecimal representation of the byte. The recommended encoding scheme to use is UTF-8. However, for compatibility reasons, if an encoding is not specified, then the default encoding of the platform is used.
+
+
+        // First calculate all the bytes needing re-encode (these will require 3 characters per instead of one in the
+        // new array)
+        for (int i = 0; i < byteArray.length; i++) {
+            if (!Character.isAlphabetic(byteArray[i]) &&
+                    !Character.isDigit(byteArray[i]) &&
+                    byteArray[i] != '.' &&
+                    byteArray[i] != '-' &&
+                    byteArray[i] != '*' &&
+                    byteArray[i] != '_' &&
+                    byteArray[i] != ' ') bytesNeedingReEncode++;
+        }
+
+
+        // Now allocate the new byte array to accomodate for the values, each encoded byte is 3 bytes now, but we already
+        // have one byte of the three for the original bytes, so we only need to allocate origlength + bytesNeedingReEncode * 2
+        byte [] encoded = new byte[byteArray.length + (bytesNeedingReEncode * 2)];
+
+        int j = 0;
+        for (int i = 0; i < byteArray.length; i++)
+        {
+            if (Character.isAlphabetic(byteArray[i]) ||
+                    Character.isDigit(byteArray[i]) ||
+                    byteArray[i] == '.' ||
+                    byteArray[i] == '-' ||
+                    byteArray[i] == '*' ||
+                    byteArray[i] == '_') {
+
+                encoded[j++] = byteArray[i];
+            } else if (byteArray[i] == ' ') {
+                encoded[j++] = '+';
+            } else {
+                encoded[j++] = '%';
+                encoded[j++] = toHexDigit((byte)((byteArray[i] & 0xf0) >> 4));
+                encoded[j++] = toHexDigit((byte)(byteArray[i] & 0x0f));
+            }
+
+            byteArray[i] = 0;
+        }
+
+        return encoded;
+    }
+
+    private static byte toHexDigit(byte b) {
+        if (b > 9) {
+            return (byte)('a' + (b - 10));
+        } else {
+            return (byte)('0' + b);
+        }
+    }
+
+
+
+
+    public static byte[] asByteArray(Object property) throws UnsupportedEncodingException, IllegalArgumentException {
+        if (property instanceof byte[])
```

Comment:
Yes, while it's unlikely we'll ever see one getBytes() can throw a UEE, and we can either choose to catch it here and recast as an IllegalArgumentException or we can bubble it up to the caller. It's cleaner to bubble it up to the caller since the caller is the one that could handle that condition effectively.